shipyard-controller/db/mongodb_state_repo.go

```diff
@@ -116,7 +117,11 @@ func (mdbrepo *MongoDBStateRepo) getSearchOptions(filter models.StateFilter) bso
 	}
 
 	if filter.KeptnContext != "" {
-		searchOptions["shkeptncontext"] = filter.KeptnContext
+		splitContexts := strings.Split(strings.ReplaceAll(filter.KeptnContext, " ", ""), ",")
```

Comment:
how can `filter.KeptnContext` contain spaces? or did you just want to trim away leading/trailing spaces? in that case there's `string.TrimSpace()`

Reply:
this was added purely to make me happy 🍡

Reply:
in theory it shouldn't but it does not harm as an additional safety measure

Reply:
although I still think we should keep it, because the filter criteria might look like the following:

```
&keptnContext="context-id1, context-id2, context-id3"
```

in that case removing all whitespaces (TrimSpace won't suffice here) is necessary since the query likely won't work otherwise. I'll add a unit test to check that

Reply:
yeah, you're right, let's keep it.
+1 for the unit test